packages/realm-server/billing/billing-queries.ts

```diff
@@ -316,3 +329,45 @@ export async function markStripeEventAsProcessed(
     param(stripeEventId),
   ]);
 }
+
+export async function getSubscriptionByStripeSubscriptionId(
+  dbAdapter: DBAdapter,
+  stripeSubscriptionId: string,
+): Promise<Subscription | null> {
+  let results = await query(dbAdapter, [
+    `SELECT * FROM subscriptions WHERE stripe_subscription_id = `,
+    param(stripeSubscriptionId),
+  ]);
+
+  if (results.length === 0) {
+    return null;
+  }
+
+  return {
+    id: results[0].id,
+    userId: results[0].user_id,
+    planId: results[0].plan_id,
+    status: results[0].status,
+  } as Subscription;
+}
+
+export async function updateSubscription(
+  dbAdapter: DBAdapter,
+  subscriptionId: string,
+  params: {
+    status: string;
+    endedAt?: number;
+  },
+) {
+  let { valueExpressions, nameExpressions: _nameExpressions } = asExpressions({
+    status: params.status,
+    ended_at: params.endedAt,
+  });
+
+  await query(dbAdapter, [
+    `UPDATE subscriptions SET (status, ended_at) = `,
```

Comment:
let's not throw away `nameExpressions`

Reply:
Sorry for this again, I am aware of this anti-pattern which we already discovered in my previous PR–I think I branched this off the other PR's branch before those `nameExpressions` fixes and it looks like I missed this spot. Will fix